pkg/cluster/service.go

```diff
@@ -126,16 +123,6 @@ func (s *ToolchainClusterService) addToolchainCluster(log logr.Logger, toolchain
 		Client:        cl,
 		ClusterStatus: &toolchainCluster.Status,
 	}
-	if cluster.Type == "" {
-		cluster.Type = Member
-	}
-	if cluster.OperatorNamespace == "" {
-		if cluster.Type == Host {
-			cluster.OperatorNamespace = defaultHostOperatorNamespace
-		} else {
-			cluster.OperatorNamespace = defaultMemberOperatorNamespace
-		}
-	}
```

Comment:
Let's fail if the operatorNamespace is not set (just to be on the safe side :-) :
```
if cluster.OperatorNamespace == "" {
	return fmt.Errorf("the operatorNamespace is not set in the ToolchainCluster CR")
}
```

Reply:
This will always throw the error making the unit tests fail as we are not setting the operator namespace here the operator namespace is going to be empty ,  https://github.com/codeready-toolchain/toolchain-common/blob/b32711b41532170d5a241bf61fce363b23b5d16c/controllers/toolchaincluster/healthchecker_test.go#L37-L98 ,  https://github.com/codeready-toolchain/toolchain-common/blob/b32711b41532170d5a241bf61fce363b23b5d16c/controllers/toolchaincluster/toolchaincluster_controller_test.go#L15-L88,  https://github.com/codeready-toolchain/toolchain-common/blob/b32711b41532170d5a241bf61fce363b23b5d16c/pkg/cluster/service_whitebox_test.go#L19-L125,   https://github.com/codeready-toolchain/toolchain-common/blob/b32711b41532170d5a241bf61fce363b23b5d16c/pkg/cluster/service_test.go#L19-L67,  https://github.com/codeready-toolchain/toolchain-common/blob/b32711b41532170d5a241bf61fce363b23b5d16c/pkg/socialevent/socialevent_name_generator_test.go#L24-L34

Reply:
Then just update the unit tests so they match the modified production code. Make sure that the label is set when it's expected to be set.

Reply:
yes, we should fail if the operator namespace is empty. The unit tests should be updated so they match the expected behavior.